a/unit-tests/main.lisp

```diff
@@ -32,41 +32,70 @@ If it does, then it is a test package."
-                                              (+ 10 (length s)))))
-                         lbge-packages))
-        lbge-packages)))
+(defun collect-all-packages ()
+  (delete-if-not #'test-package-p (list-all-packages)))
 
 (defun delete-test-packages ()
-  (let ((all-packages (collect-all-packages)))
-    (mapcar #'delete-package all-packages)))
+  "Find and delete all test packages"
+  (let ((all-test-packages (collect-all-packages)))
+    (mapcar #'delete-package all-test-packages)))
 
-(defun collect-test-packages (&optional selected-package)
-  (filter-disabled-packages (collect-all-packages selected-package)))
+(defun collect-active-tests ()
+  (filter-disabled-packages (collect-all-packages)))
 
 (defun test-package-disabled-p (package)
   (find-symbol "*LBGE-SKIP-TEST*" package))
 
 (defun filter-disabled-packages (package-list)
   (delete-if #'test-package-disabled-p package-list))
 
-(defun run (&key (reporter :spec) selected-package)
+(defun run-selected-test-suite (packages selected-test-suite)
+  (let ((suite (find-package selected-test-suite)))
+    (if suite
+      (rove:run-suite suite)
+      (format t "Could not run test suite: package ~A not found~%" selected-test-suite))))
+
+(defun report-results ()
+  (let ((failed (rove/core/stats:stats-failed rove:*stats*))
+        (passed (rove/core/stats:stats-passed rove:*stats*)))
+
+    (when (> (length passed) 0)
+      (format t "~%-------------~%Passed tests:~%")
+      (loop :for test :across (rove/core/stats:stats-passed rove:*stats*) :do
+        (format t "* ~A~%" (rove/core/result:test-name test))))
+    (when (> (length failed) 0)
+      (format t "~%-------------~%Failed tests:~%")
+      (loop :for test :across failed :do
+        (format t "* ~A~%" (rove/core/result:test-name test))))
+    (when (= (length failed) 0)
+      (format t "~%All tests passed successfully!~%"))
+    (= 0 (length failed))))
+
+(defun print-start-message ()
+  (terpri)
+  (terpri)
+  (terpri)
+  (princ ",---------------------------------------.") (terpri)
+  (princ "| :~-<==={ Starting unit tests }===>-~: |") (terpri)
+  (princ "`---------------------------------------'") (terpri)
+  (terpri))
+
+(defun run (&key (reporter :spec) selected-test-suite)
+  "Run unit tests and report the result.
+`reporter`: rove reporter. Default is :spec
+`selected-test-suite`: if present, only test suite with this name will run.
+Name must be string designator of the full suite name, e.g. `:lbge.test.engine'"
   (delete-test-packages)
   (load-test-files)
   (rove:use-reporter reporter)
-  (let* ((all-packages (collect-test-packages selected-package))
+  (let* ((all-packages (collect-active-tests))
          (filtered-packages (filter-disabled-packages all-packages)))
-    (format t "filtered packages: ~S~%" filtered-packages)
+    (when selected-test-suite
+        (run-selected-test-suite filtered-packages selected-test-suite)
+        (return-from run (report-results)))
```

Comment:
It's a small thing, but maybe add newlines around this `when` block to indicate that it's a different thing from the others and returns from the function.